### PR TITLE
version 2 mobile endpoints for retrieving keep data (regarding parsing h...

### PR DESCRIPTION
**Pull request by ahsu1230**
...ashtags out of notes)

@2is10 @andrewconner

### CODE REVIEW

**Comment by ahsu1230** [comment]
next PR will be a new mobile endpoint to edit keep notes (works just like website & ext)

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileKeepsController.scala [hunk 1]
```diff
@@ -39,17 +39,25 @@ class MobileKeepsController @Inject() (
...
-    keepsCommander.allKeeps(before map ExternalId[Keep], after map ExternalId[Keep], collectionOpt map ExternalId[Collection], helprankOpt, count, request.userId) map { res =>
+  def allKeepsV1(before: Option[String], after: Option[String], collectionOpt: Option[String], helprankOpt: Option[String], count: Int, withPageInfo: Boolean) = UserAction.async { request =>
+    getAllKeeps(request.userId, before, after, collectionOpt, helprankOpt, count, withPageInfo, true)
+  }
+
+  def allKeepsV2(before: Option[String], after: Option[String], collectionOpt: Option[String], helprankOpt: Option[String], count: Int, withPageInfo: Boolean) = UserAction.async { request =>
+    getAllKeeps(request.userId, before, after, collectionOpt, helprankOpt, count, withPageInfo, false)
+  }
+
+  private def getAllKeeps(userId: Id[User], before: Option[String], after: Option[String], collectionOpt: Option[String], helprankOpt: Option[String], count: Int, withPageInfo: Boolean, parseKeepNote: Boolean) = {
```
Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileKeepsController.scala]: might consider renaming the `parseKeepNote` parameters to `v1` (throughout this pull request)

Comment (ahsu1230) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileKeepsController.scala]: sure, I could do that. it's shorter & more general anyway lol

**Comment by 2is10** [comment]
:+1:

**Comment by ahsu1230** [comment]
@thassss @mrbrown14 @jaredpetker @DerekBurch 
When you guys are ready to support hashtags in notes, please update the following endpoints:

```
GET     /m/2/libraries/:id                         (gets first few keeps of a library)
GET     /m/2/libraries/:id/keeps              (gets paginated keeps)
GET     /m/2/keeps/:id                            (gets a keep info)
GET     /m/2/keeps/all                            (gets keeps for a tag, for example)
POST  /m/2/libraries                              (gets writable libraries for a url, with keep data for each library url is kept to)
```

**Pull request merged**